Tests/Unit/MSFT_xDefaultGatewayAddress.Tests.ps1

```diff
@@ -50,193 +80,227 @@ InModuleScope MSFT_xDefaultGatewayAddress {
             }
+        }
+
+        Context 'checking return with no gateway but one is currently set' {
+            It 'should return false' {
 
-            It 'when default gateway does not exist should be $true' {
                 $Splat = @{
                     InterfaceAlias = 'Ethernet'
                     AddressFamily = 'IPv4'
                 }
-                $Result = Test-Properties @Splat
-                $Result | Should Be $true
-            }
-
-            It 'should call Get-NetRoute twice' {
-                Assert-MockCalled -commandName Get-NetRoute -Exactly 2
+                Test-TargetResource @Splat | Should Be $False
             }
         }
 
-        Context 'invoking with -Apply switch and default gateway exists' {
+        #region Mocks
+        Mock Get-NetRoute -MockWith {}
+        #endregion
 
-            #region Mocks
-            Mock Get-NetRoute -MockWith {
-                [PSCustomObject]@{
-                    NextHop = '192.168.0.1'
-                    DestinationPrefix = '0.0.0.0/0'
+        Context 'checking return with default gateway but none are currently set' {
+            It 'should return false' {
+
+                $Splat = @{
+                    Address = '192.168.0.1'
                     InterfaceAlias = 'Ethernet'
-                    InterfaceIndex = 1
                     AddressFamily = 'IPv4'
                 }
+                Test-TargetResource @Splat | Should Be $False
             }
-            Mock New-NetRoute
-            Mock Remove-NetRoute
-            #endregion
+        }
+
+        Context 'checking return with no gateway and none are currently set' {
+            It 'should return true' {
 
-            It 'when default gateway does not match should be $true' {
                 $Splat = @{
-                    Apply = $true
-                    Address = '10.0.0.2'
                     InterfaceAlias = 'Ethernet'
                     AddressFamily = 'IPv4'
                 }
-                $Result = Test-Properties @Splat
-                $Result | Should Be $true
+                Test-TargetResource @Splat | Should Be $True
             }
+        }
+    }
+
+    #######################################################################################
+
+    Describe 'Test-ResourceProperty' {
 
-            It 'when default gateway matches should be $true' {
+        Mock Get-NetAdapter -MockWith { [PSObject]@{ Name = 'Ethernet' } }
+
+        Context 'invoking with bad interface alias' {
+
+            It 'should throw an error' {
                 $Splat = @{
-                    Apply = $true
                     Address = '192.168.0.1'
-                    InterfaceAlias = 'Ethernet'
+                    InterfaceAlias = 'NotReal'
                     AddressFamily = 'IPv4'
                 }
-                $Result = Test-Properties @Splat
-                $Result | Should Be $true
+                { Test-ResourceProperty @Splat } | Should Throw
```

Comment:
Ah good stuff! Didn't know you could do that (specify the error message expected from the throw). The hash table makes a lot of sense as well. I'll get it done in this PR later on today! Thanks heaps for the feedback - very helpful!